lms/static/js/student_account/components/PasswordResetConfirmation.jsx

```diff
@@ -0,0 +1,140 @@
+/* globals gettext */
+
+import 'whatwg-fetch';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { Button, StatusAlert } from '@edx/paragon/static';
+
+import PasswordResetInput from './PasswordResetInput';
+
+// NOTE: Use static paragon with this because some internal classes (StatusAlert at least)
+// conflict with some standard LMS ones ('alert' at least). This means that you need to do
+// something like the following on any templates that use this class:
+//
+// <link type='text/css' rel='stylesheet' href='${STATIC_URL}paragon/static/paragon.min.css'>
+//
+
+class PasswordResetConfirmation extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      password1: '',
+      password2: '',
+      passwordsMatch: true,
+      isValid: true,
+      validationMessage: '',
+    };
+    this.onBlurPassword1 = this.onBlurPassword1.bind(this);
+    this.onBlurPassword2 = this.onBlurPassword2.bind(this);
+  }
+
+  onBlurPassword1(password) {
+    this.updatePasswordState(password, this.state.password2);
+    this.validatePassword(password);
+  }
+
+  onBlurPassword2(password) {
+    this.updatePasswordState(this.state.password1, password);
+  }
+
+  updatePasswordState(password1, password2) {
+    this.setState({
+      password1,
+      password2,
+      passwordsMatch: !password1 || !password2 || (password1 === password2),
+    });
+  }
+
+  validatePassword(password) {
+    fetch('/api/user/v1/validation/registration', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        password,
+      }),
+    })
+    .then(res => res.json())
```

Comment:
Fetch doesn't reject [`4xx` and `5xx` errors](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Checking_that_the_fetch_was_successful) - you need to check `response.ok` explicitly.

![image](https://user-images.githubusercontent.com/8136030/38372293-7479a330-38bc-11e8-8558-d8698efbbb32.png)

Reply:
I know, but I don't care about specially handling errors here I don't think? I tried to write it such that we only show a validation error if the response has the data. And if there's any problem getting that from the response, we don't change anything.

Reply:
Yeah, I don't think it'll have an impact - maybe it's worth just throwing in an empty `catch` but 🤷‍♂️ (do you know if we have any client-side error logging to `NewRelic` ? I'm guessing the answer is probably no...)